Library/Homebrew/utils/analytics.rb

```diff
@@ -1,34 +1,52 @@
-
-def analytics_anonymous_prefix_and_os
+def analytics_label
   @analytics_anonymous_prefix_and_os ||= begin
-    "#{OS_VERSION}, #{HOMEBREW_PREFIX.to_s.gsub(ENV["HOME"], "~")}"
+    os = OS_VERSION
+    prefix = if HOMEBREW_PREFIX.to_s == "/usr/local"
+      "/usr/local"
+    else
+      "non-/usr/local"
+    end
+    ci = ", CI=1" if ENV["CI"]
+    "#{os}, #{prefix}#{ci}"
   end
 end
 
 def report_analytics(type, metadata={})
   return unless ENV["HOMEBREW_ANALYTICS"]
+  return if ENV["HOMEBREW_NO_ANALYTICS"]
+
+  debug = !!ENV["HOMEBREW_ANALYTICS_DEBUG"]
 
-  metadata_args = metadata.map do |key, value|
+  args = metadata.map do |key, value|
     ["-d", "#{key}=#{value}"] if key && value
   end.compact.flatten
 
+  url = if !debug
```

Comment:
@bfontaine That's intentional because I think it's easier to parse than `unless..else`